pkg/mimirtool/commands/runtimeconfig.go

```diff
@@ -0,0 +1,60 @@
+package commands
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/flagext"
+	"github.com/grafana/dskit/runtimeconfig"
+
+	"github.com/grafana/mimir/pkg/mimir"
+)
+
+// RuntimeConfigCommand works with Mimir's runtime config.
+type RuntimeConfigCommand struct {
+	configFile string
+}
+
+// Register runtime config related commands and flags with the kingpin application.
+func (c *RuntimeConfigCommand) Register(app *kingpin.Application, _ EnvVarNames) {
+	cmd := app.Command("runtime-config", "Work with Grafana Mimir runtime configuration.")
+	validateCmd := cmd.Command("validate", "Validate a runtime configuration file").Action(c.validateConfig)
+
+	validateCmd.Flag("config-file", "The runtime configuration file to validate.").StringVar(&c.configFile)
+}
+
+func (c *RuntimeConfigCommand) validateConfig(*kingpin.ParseContext) error {
```

Comment:
Nit:
```suggestion
	validateCmd := cmd.Command("validate", "Validate a runtime configuration file").Action(c.validateCommand)

	validateCmd.Flag("config-file", "The runtime configuration file to validate.").StringVar(&c.configFile)
}

func (c *RuntimeConfigCommand) validateCommand(*kingpin.ParseContext) error {
```

Reply:
It's not the norm to give action methods the `Command` suffix though. See for example the `remote-read` [sub-commands](https://github.com/grafana/mimir/blob/main/pkg/mimirtool/commands/remote_read.go#L55-L57). To me `validateCommand` sounds as if the method validates a command.

Reply:
Yeah, I was hesitant to just call it `validate` or `validateCommand`. I opted for the latter because the variable name it yields is called `validateCmd` and after all, it's the _"validate" command_.

I'm okay renaming it to `validate` too, but `validateConfig` is inconsistent with the command name which is `validate` (it calling it validateConfig is redundant because it's part of `runtime-config` already.)

Reply:
OK, then I understand what you mean :)

Reply:
I named it `validate`, since we don't use a `Command` suffix otherwise.